controllers/apps/operations/rebuild_instance.go

```diff
@@ -91,38 +73,92 @@ func (r rebuildInstanceOpsHandler) ActionStartedCondition(reqCtx intctrlutil.Req
 }
 
 func (r rebuildInstanceOpsHandler) Action(reqCtx intctrlutil.RequestCtx, cli client.Client, opsRes *OpsResource) error {
-	if opsRes.OpsRequest.Spec.Force {
-		return nil
-	}
 	for _, v := range opsRes.OpsRequest.Spec.RebuildFrom {
 		compStatus, ok := opsRes.Cluster.Status.Components[v.ComponentName]
 		if !ok {
 			continue
 		}
 		// check if the component has matched the `Phase` condition
-		if !slices.Contains([]appsv1alpha1.ClusterComponentPhase{appsv1alpha1.FailedClusterCompPhase,
+		if !opsRes.OpsRequest.Spec.Force && !slices.Contains([]appsv1alpha1.ClusterComponentPhase{appsv1alpha1.FailedClusterCompPhase,
 			appsv1alpha1.AbnormalClusterCompPhase, appsv1alpha1.UpdatingClusterCompPhase}, compStatus.Phase) {
 			return intctrlutil.NewFatalError(fmt.Sprintf(`the phase of component "%s" can not be %s`, v.ComponentName, compStatus.Phase))
 		}
+		var (
+			synthesizedComp *component.SynthesizedComponent
+			err             error
+			instanceNames   []string
+		)
 		for _, ins := range v.Instances {
 			targetPod := &corev1.Pod{}
 			if err := cli.Get(reqCtx.Ctx, client.ObjectKey{Name: ins.Name, Namespace: opsRes.Cluster.Namespace}, targetPod); err != nil {
 				return err
 			}
-			synthesizedComp, err := r.buildSynthesizedComponent(reqCtx, cli, opsRes.Cluster, targetPod.Labels[constant.KBAppComponentLabelKey])
+			synthesizedComp, err = r.buildSynthesizedComponent(reqCtx, cli, opsRes.Cluster, targetPod.Labels[constant.KBAppComponentLabelKey])
 			if err != nil {
 				return err
 			}
-			isAvailable, _ := r.instanceIsAvailable(synthesizedComp, targetPod, "")
-			if isAvailable {
+			isAvailable, _ := instanceIsAvailable(synthesizedComp, targetPod, "")
+			if !opsRes.OpsRequest.Spec.Force && isAvailable {
 				return intctrlutil.NewFatalError(fmt.Sprintf(`instance "%s" is availabled, can not rebuild it`, ins.Name))
 			}
+			instanceNames = append(instanceNames, ins.Name)
+		}
+		if !v.InPlace {
+			if synthesizedComp.Name != v.ComponentName {
```

Comment:
null pointer if v.Instances is empty

Reply:
v.instances is required in the api.

Reply:
could be 
```
rebuildFrom:
    instances: []
```

Reply:
fixed, add `+kubebuilder:validation:MinItems=1` check